### PR TITLE
DDF-1006 Ensure that default values get applied for the STS Server.

### PR DESCRIPTION
**Pull request by brendan-hofmann**

### CODE REVIEW

**Comment by shaundmorris** [comment]
Looks good to me.

**Comment by shaundmorris** [comment]
If there are no objections, I would like to merge this PR today.

File: sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java [hunk 1]
```diff
@@ -45,10 +43,15 @@ public PropertyPlaceholderWrapper(DefaultConditionsProvider conditionsProvider,
             StaticSTSProperties properties) {
         samlConditionsProvider = conditionsProvider;
         stsProperties = properties;
+        // set the default values in case there is no configuration
+        samlConditionsProvider.setLifetime(1800);
```
Comment (mrmateo) [inline sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java]: Just curious, would it be worthwhile to check if there are already values in the samlConditionsProvider and stsProperties before setting them to default values? Not sure if that's something that could happen (i.e. values passed in to the constructor).

File: sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java [hunk 2]
```diff
@@ -45,10 +43,15 @@ public PropertyPlaceholderWrapper(DefaultConditionsProvider conditionsProvider,
...
+        stsProperties.setSignatureUsername("localhost");
+        stsProperties.setIssuer("localhost");
+        stsProperties.setEncryptionUsername("localhost");
```
Comment (mrmateo) [inline sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java]: Just an FYI -- I am pretty sure the "localhost" and 1800 values will be marked as code violations (duplicate strings and magic numbers) in tools like SONAR although they aren't necessary issues. Not a big deal, but just something to keep in mind.

File: sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java (hunk 1 above)
Comment (brendan-hofmann) [inline sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java]: The way that it should work is that the constructor initializes the default values on bundle startup. If there are any existing configuration values, then the config handler method would get called immediately afterwards they would be updated with the correct values.

I suppose we could add the config to the constructor as well, but then we'd have to deal with some more null checking and conditional logic. That seems unnecessary to me, but could be done if you think it necessary.

File: sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java (hunk 2 above)
Comment (brendan-hofmann) [inline sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java]: I could refactor them into constants for the class. Unfortunately this is how it was with the property-placeholder,  essentially. There's not really anywhere that I've found where these default values could be pulled from as constants.

Stylistically I could go either way, I'm fine with constants too.

Comment (mrmateo) [inline sts/security-sts-server/src/main/java/ddf/security/sts/PropertyPlaceholderWrapper.java]: yeah, private class constants should be good.

**Comment by shaundmorris** [comment]
Yeah I thought about "localhost" being constants, but didn't realize they were violations so that is nice to clean-up

**Comment by mrmateo** [comment]
Looks good to me, thanks @brendan-hofmann !

**Pull request merged**